### PR TITLE
(PA-1632) Update augeas to 1.10.1

### PR DESCRIPTION
**Pull request by caseywilliams**
Updates augeas to 1.10.1 - I had to make a few adjustments to get this to compile:

- Added a manually created pkgconfig file for libselinux on rhel5, where we use an older version of that library that doesn't seem to create one (but augeas 1.10.1 wants one).
- Patched augeas's gnulib on solaris 10 sparc - the gnulib files aren't created until the configure step, so the patch is applied there. To be frank, I don't entirely understand the exact nature of the problem, but what I've done here _seems_ to fix it. Suggestions about more correct ways to do this are 110% welcome.

[This puppet-runtime build](http://jenkins-release.delivery.puppetlabs.net/job/vanagon_generic_job/1382/) shows the original errors for rhel5 and solaris 10 sparc (we aren't merging puppet-runtime until after the 5.4.0 release, so I've ported the changes to this repo). Master jenkins is going down for maintenance right now, but I'll trigger PR tests here once it's back up.

### CODE REVIEW

**Comment by caseywilliams** [comment]
jenkins please test this on aix61-POWERfa,aix71-POWERfa,arista4-32a,ciscoxr-64a,cumulus25-64a,debian7-64a,debian7-32a,debian8-64a,debian8-32a,debian9-64a,debian9-32a,osx1011-64a,redhat5-64a,redhat5-32a,redhat7-AARCH64a,sles11-64a,sles12-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-SPARCa,ubuntu1404-64a,ubuntu1604-POWERa,windows2016-64a

**Comment by caseywilliams** [comment]
jenkins please test this on aix61-POWERfa,aix71-POWERfa,arista4-32a,ciscoxr-64a,cumulus25-64a,debian7-64a,debian8-32a,osx1011-64a,redhat5-64a,redhat5-32a,redhat7-AARCH64a,sles11-64a,sles12-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-SPARCa,ubuntu1404-64a,ubuntu1604-POWERa,windows2016-64a

**Comment by caseywilliams** [comment]
jenkins please test this on aix61-POWERfa,aix71-POWERfa,arista4-32a,ciscoxr-64a,cumulus25-64a,debian7-64a,debian8-32a,osx1011-64a,redhat5-64a,redhat5-32a,redhat7-AARCH64a,sles11-64a,sles12-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-SPARCa,ubuntu1404-64a,ubuntu1604-POWERa,windows2016-64a

**Comment by caseywilliams** [comment]
Ok, acceptance tests are passing now

**Comment by ScottGarman** [comment]
Does augeas have a configure option to specify the location of libselinux? That would be preferable to generating a pkgconfig file.

File: configs/components/augeas.rb [hunk 1]
```diff
@@ -89,6 +86,16 @@
     ["./configure --prefix=#{settings[:prefix]} #{settings[:host]}"]
   end
 
+  if platform.name =~ /solaris-10-sparc/
+    # Augeas builds its own gnulib during configure -
+    # This patch to gnulib fixes a linking error around symbol versioning in pthread
+    pkg.add_source 'file://resources/patches/augeas/augeas-1.10.1-gnulib-pthread-in-use.patch'
```
Comment (branan) [inline configs/components/augeas.rb]: Why can't we just use the `apply_patch` method here?

Comment (caseywilliams) [reply configs/components/augeas.rb]: gnulib comes from a submodule that's cloned during the configure step, so the files don't exist right away - I'm not aware of a way to accomplish this with vanagon

**Comment by caseywilliams** [comment]
@ScottGarman it looks like configure offers a `--without-selinux` option which just disables the use of selinux altogether even on platforms that have it - I don't see any other options that are specifically selinux-related, though.

File: configs/components/augeas.rb (hunk 1 above)
Comment (caseywilliams) [reply configs/components/augeas.rb]: @branan oh, or would it work if I just called `pkg.apply_patch` within the configure block here? I didn't try that.

Comment (branan) [reply configs/components/augeas.rb]: That sounds hacky. I'm fine just doing it manually, but would like a comment better explaining what's up

**Comment by ScottGarman** [comment]
@caseywilliams ok, that being the case I'm ok with shipping a pkgconfig file with libselinux. I'm 👍 once Branan's concerns have been addressed.

**Comment by branan** [comment]
I guess my only other concern is that this is a pretty huge bump to do in 1.10 - has anyone gone through the augeas changelog and tried to do any sort of risk analysis on this?

**Comment by caseywilliams** [comment]
@branan agreed, and no, not that I know of -- should I retarget?

**Comment by branan** [comment]
@caseywilliams it looks like the ticket was targetted at 5.4.0, so that would be master - but if I missed some discussion to land it earlier that's fine

**Comment by caseywilliams** [comment]
Oh wow, @branan I completely missed that - I'll rebase.

**Comment by caseywilliams** [comment]
jenkins please test this on redhat5-64a,solaris10-SPARCa

**Comment by caseywilliams** [comment]
Alright, this is retargeted at master and passed acceptance. I think it's ready for merge unless anyone has improvements to suggest.

**Pull request merged**